catalog/app/utils/Config.js

```diff
@@ -0,0 +1,173 @@
+import PT from 'prop-types';
+import * as R from 'ramda';
+import * as React from 'react';
+import * as RC from 'recompose';
+
+import AsyncResult from 'utils/AsyncResult';
+import Data from 'utils/Data';
+import { BaseError } from 'utils/error';
+import * as RT from 'utils/reactTools';
+import { conforms, isNullable, isArrayOf } from 'utils/validate';
+
+
+export class ConfigError extends BaseError {
+  static displayName = 'ConfigError'
+}
+
+const parseJSON =
+  R.tryCatch(JSON.parse, (e, src) => {
+    throw new ConfigError(`invalid JSON:\n${src}`, { src, originalError: e });
+  });
+
+const validateConfig = conforms({
+  alwaysRequiresAuth: R.is(Boolean),
+  sentryDSN: isNullable(String),
+  apiGatewayEndpoint: R.is(String),
+  defaultBucket: R.is(String),
```

Comment:
@meffij the idea of using listBuckets didnt work out: this method doesn't support CORS, so we're back to listing the default bucket contents.
also, i've added explicit sign-in / sign-out redirects to the config